### PR TITLE
[UX] VirtualKeyPopup: respond to hold & pan release

### PR DESCRIPTION
**Pull request by Frenzie**
As suggested by @poire-z in https://github.com/koreader/koreader/pull/4886#discussion_r273569802

This is actually much better than before.

It subtly changes the behavior of every key this way, but I don't think that's a bad thing.

### CODE REVIEW

**Comment by Frenzie** [comment]
@poire-z Don't merge, the behavior isn't 100 % right yet.

**Comment by Frenzie** [comment]
I'll go ahead and merge this one 'cause it's such a definite improvement. ;-)

File: frontend/ui/widget/virtualkeyboard.lua [hunk 1]
```diff
@@ -298,11 +313,21 @@ function VirtualKeyPopup:init()
...
+
+                    -- restore ability to hold/pan release on central key after opening popup
+                    virtual_key._keyOrigHoldPanHandler = function()
+                        virtual_key.onHoldReleaseKey = virtual_key._onHoldReleaseKey
+                        virtual_key.onPanReleaseKey = virtual_key._onPanReleaseKey
+                    end
+                    virtual_key._onHoldReleaseKey = virtual_key.onHoldReleaseKey
+                    virtual_key.onHoldReleaseKey = virtual_key._keyOrigHoldPanHandler
+                    virtual_key._onPanReleaseKey = virtual_key.onPanReleaseKey
+                    virtual_key.onPanReleaseKey = virtual_key._keyOrigHoldPanHandler
```
Comment (poire-z) [inline frontend/ui/widget/virtualkeyboard.lua]: What does all that allow?
Not really understanding - looks to me you hold on the original keyboard key - and this virtual key (even with the same letter) is another VirtualKey object, so when you release that second one, the first one should still work. (?)

Comment (Frenzie) [reply frontend/ui/widget/virtualkeyboard.lua]: This doesn't allow anything. ;-) It restricts the central key in the popup from responding as if there were an intentional hold release after opening. But after opening, it felt odd when it didn't respond like any other key.

**Comment by poire-z** [comment]
Some small remarks:
- On my Kobo, the popup keys are not vertically aligned with the main keyboard keys, while they are on the emulator. So, may be a missing scaleBySize() somewhere (or just me having missed updating my small private hacks in that file for this change, so please check on your Kobo :)
- With flash keyboard enabled, when taping another key in the popup, the taped key is border-flashed. But when panning and releasing on another key, it's the middle key that is border-flashed, and not the selected other key.

**Comment by Frenzie** [comment]
If by not vertically aligned you mean off by a pixel or two then it's probably just something like a wrong number of subtracted borders or something. If you truly mean not vertically aligned then I'm fairly sure it's just you. ;-)

**Comment by poire-z** [comment]
Yeah, I mean off by two pixels or three :)

Other remarks:
- Dunno if that should happen on popup key tap, but may be on popup key panrelease, the popup could close, as it happens on some(all?) android virtual keyboards - to avoid that additional tap to close it
- Possibly, for the extra keys definitions (`local _e_ =`), it would be nice if they were put into an independant file, so we can import them easily into the other layouts (fr_keyboard.lua), without having to redefine them all (even if of fr, we may tweak a few of them to get `ê` at north). Not sure if there's a way in Lua like in python: `from extra_keys import *`.

**Pull request merged**